### DISCUSSION TITLE
Bring drawRect back

### DISCUSSION THREAD

**Discussion by luanpotter** [flame-engine/flame#625] [Ideas]
@erickzanardo convinced me we should have two overloads for the render method (the current one and the one that takes a Rect)

**Comment by spydon** [comment]
Can you explain why it is needed?

**Reply by erickzanardo** [reply]
Rect is a really useful class, I have caught myself creating rects to do some operation and then turning then back to Vector2 to render, so this is a good utility method.

**Reply by spydon** [reply]
Could we add those operations on Vector2, or are there operations involving code we are not in control of?

**Reply by erickzanardo** [reply]
That is a good question, we sure could try enhance Vector2 and see how it goes, the problem for me is that Rect and Vector2 are not equivalent, like, Rect has a position and a size, a Vector2, only the position.

I will list some of the methods that I commonly use on Rect and we can try to think if that could be applied to Vector2:

 - inflate
 - deflate
 - shift
 - translate
 - fromCenterPoint (constructor)

We could also maybe add some extensions on canvas for vector2, like a `drawVector2Rect(pos, size, paint)`, that could help as well.
